src/language/typescript/common/normalized-name.ts

```diff
@@ -0,0 +1,23 @@
+import { Endomorphism } from 'fp-ts/lib/function';
+import { Branded } from 'io-ts';
+import { Ref } from '../../../utils/ref';
+
+export interface NormalizedNameBrand {
+	readonly NormalizedName: unique symbol;
+}
+
+export type NormalizedName = Branded<string, NormalizedNameBrand>;
+
+const INVALID_NAMES = ['Error', 'Promise', 'PromiseLike', 'Array', 'ArrayLike', 'Function', 'Object'];
+
+export const makeNormalizedName = (name: string): NormalizedName => {
+	return (INVALID_NAMES.includes(name) ? `${name}Type` : name) as NormalizedName;
```

Comment:
Please, no casts. If you are sure you can build `NormalizedName` from string then there're no benefits in using brand.